crates/noirc_evaluator/src/ssa_refactor/ssa_builder/mod.rs

```diff
@@ -152,44 +161,52 @@ impl FunctionBuilder {
+    /// Returns the block currently being modified
     pub(crate) fn current_block(&mut self) -> BasicBlockId {
         self.current_block
     }
 
-    /// Insert an allocate instruction at the end of the current block, allocating the
-    /// given amount of field elements. Returns the result of the allocate instruction,
+    /// Insert an allocate instruction at the end of the current block.
+    /// Returns the result of the allocate instruction,
     /// which is always a Reference to the allocated data.
     pub(crate) fn insert_allocate(&mut self) -> ValueId {
+        // TODO: Rust has .first on vectors which is confusing because
+        // TODO that returns Option<T> whereas this returns T and expects there
+        // TODO to be only one T.Perhaps change this to be `first_and_only`
+        // TODO or something to not conflate this with .first, but also
+        // TODO to convey the fact that its the only one; `exactly_one` ?
         self.insert_instruction(Instruction::Allocate, None).first()
     }
 
-    /// Insert a Load instruction at the end of the current block, loading from the given offset
-    /// of the given address which should point to a previous Allocate instruction. Note that
-    /// this is limited to loading a single value. Loading multiple values (such as a tuple)
-    /// will require multiple loads.
-    /// 'offset' is in units of FieldElements here. So loading the fourth FieldElement stored in
-    /// an array will have an offset of 3.
-    /// Returns the element that was loaded.
+    /// Insert a Load instruction at the end of the current block.
+    /// TODO: check previous description was not outdated.
+    /// TODO: where do we check that `address` is a Reference?
     pub(crate) fn insert_load(&mut self, address: ValueId, type_to_load: Type) -> ValueId {
         self.insert_instruction(Instruction::Load { address }, Some(vec![type_to_load])).first()
     }
 
-    /// Insert a Store instruction at the end of the current block, storing the given element
-    /// at the given address. Expects that the address points somewhere
-    /// within a previous Allocate instruction.
+    /// Insert a Store instruction at the end of the current block.
+    /// Storing the given element at the given address.
+    /// TODO: check previous description was not outdated
+    /// TODO: where do we check that `address` is a Reference?
+    /// TODO: where do we check that `value` is the correct type for `address?
+    /// TODO: Is `address` still correct terminology, given that we are using stores for mutable variables?
     pub(crate) fn insert_store(&mut self, address: ValueId, value: ValueId) {
         self.insert_instruction(Instruction::Store { address, value }, None);
     }
 
     /// Insert a binary instruction at the end of the current block.
+    ///
     /// Returns the result of the binary instruction.
+    ///
+    /// All binary instructions return one result.
```

Comment:
I don't think this comment is necessary given the comment above it already mentions it returns a singular result.